var/spack/repos/builtin/packages/geopm/package.py

```diff
@@ -0,0 +1,125 @@
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Geopm(AutotoolsPackage):
+    """GEOPM is an extensible power management framework targeting HPC.
+    The GEOPM package provides libgeopm, libgeopmpolicy and applications
+    geopmctl and geopmpolicy, as well as tools for postprocessing.
+    GEOPM is designed to be extended for new control algorithms and new
+    hardware power management features via its plugin infrastructure.
+
+    Note: GEOPM interfaces with hardware using Model Specific Registers (MSRs).
+    For propper usage make sure MSRs are made available directly or via the
+    msr-safe kernel module by your administrator."""
+
+    homepage = "https://geopm.github.io"
+    url      = "https://github.com/geopm/geopm/releases/download/v0.4.0/geopm-0.4.0.tar.gz"
+
+    # Add additional proper versions and checksums here. "spack checksum geopm"
+    version('0.4.0', 'd4cc8fffe521296dab379857d7e2064d')
+    version('0.3.0', '568fd37234396fff134f8d57b60f2b83')
+    version('master', git='https://github.com/geopm/geopm.git', branch='master')
+    version('develop', git='https://github.com/geopm/geopm.git', branch='dev')
+
+    # Variants reflecting most ./configure --help options
+    variant('debug', default=False, description='Enable debug.')
+    variant('coverage', default=False, description='Enable test coverage support, enables debug by default.')
+    variant('procfs', default=True, description='Enable procfs (disable for OSes not using procfs).')
+    variant('mpi', default=True, description='Enable MPI dependent components.')
+    variant('fortran', default=True, description='Build fortran interface.')
+    variant('doc', default=True, description='Create man pages with ruby-ronn.')
+    variant('openmp', default=True, description='Build with OpenMP.')
+    variant('ompt', default=False, description='Use OpenMP Tools Interface.')
+    variant('hwloc', default=True, description='Build with hwloc.')
+    variant('gnu-ld', default=False, description='Assume C compiler uses gnu-ld.')
+
+    # Added dependencies.
+    depends_on('m4', type='build')
+    depends_on('automake', type='build')
+    depends_on('autoconf', type='build')
+    depends_on('libtool', type='build')
+    depends_on('ruby-ronn', type='build', when='+doc')
+    depends_on('doxygen', type='build', when='+doc')
+    depends_on('numactl')
+    depends_on('mpi', when='+mpi')
+    # TODO: check if hwloc@specific-version still required with future openmpi
+    depends_on('hwloc@1.11.9', when='+hwloc')
+    depends_on('py-pandas', type='run')
+    depends_on('py-numpy', type='run')
+    depends_on('py-natsort', type='run')
+    depends_on('py-matplotlib', type='run')
+
+    parallel = False
+
+    def configure_args(self):
+        spec = self.spec
+        args = []
+
+        if '+debug' in spec:
```

Comment:
Minor note. All these conditionals can be substituted by an `AutotoolsPackage` specific feature:
```python
args.extend(self.enable_or_disable('debug'))
```
similarly for other variants.